sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -246,6 +246,31 @@ case class UpdateTableStatement(
     values: Seq[Expression],
     condition: Option[Expression]) extends ParsedStatement
 
+/**
+ * A MERGE INTO tbl_name statement, as parsed from SQL.
+ */
+case class MergeIntoStatement(
```

Comment:
do we really need a statement for v2 only commands? It looks to me that this only creates code duplication without any benefit. cc @rdblue

Reply:
I was thinking it is a convention to make the parse-analyze-plan flow more clear.